base/linalg/arpack.jl

```diff
@@ -12,20 +12,20 @@ function aupd_wrapper(T, matvecA!::Function, matvecB::Function, solveSI::Functio
                       tol::Real, maxiter::Integer, mode::Integer, v0::Vector)
     lworkl = cmplx ? ncv * (3*ncv + 5) : (sym ? ncv * (ncv + 8) :  ncv * (3*ncv + 6) )
     TR = cmplx ? T.types[1] : T
-    TOL = Array{TR}(1)
+    TOL = Vector{TR}(1)
```

Comment:
Not directly the topic of this PR, but I'm pretty sure that you can just adjust the driver functions and pass `tol` to them without wrapping. It is only an input argument so so `&` should be fine.